pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,191 @@
+	AppliedTo []NetworkPolicyPeer `json:"appliedTo"`
+	// Set of ingress rules evaluated based on the order in which they are set.
+	// Currently Ingress rule supports setting the `From` field but not the `To`
+	// field within a Rule.
+	// +optional
+	Ingress []Rule `json:"ingress"`
+	// Set of egress rules evaluated based on the order in which they are set.
+	// Currently Egress rule supports setting the `To` field but not the `From`
+	// field within a Rule.
+	// +optional
+	Egress []Rule `json:"egress"`
+}
+
+// Rule describes the traffic allowed to/from the workloads selected by
+// Spec.AppliedTo. Based on the action specified in the rule, traffic is either
+// allowed or denied which exactly match the specified ports and protocol.
+type Rule struct {
+	// Action specifies the action to be applied on the rule. Defaults to
+	// ALLOW action.
+	// +optional
+	Action *RuleAction `json:"action"`
+	// Set of port and protocol allowed/denied by the rule. If this field is unset
+	// or empty, this rule matches all ports.
+	// +optional
+	Ports []NetworkPolicyPort `json:"ports"`
+	// Rule is matched if traffic originates from workloads selected by
+	// this field. If this field is empty, this rule matches all sources.
+	// +optional
+	From []NetworkPolicyPeer `json:"from"`
+	// Rule is matched if traffic is intended for workloads selected by
+	// this field. If this field is empty or missing, this rule matches all
+	// destinations.
+	// +optional
+	To []NetworkPolicyPeer `json:"to"`
+}
+
+// NetworkPolicyPeer describes the grouping selector of workloads.
+type NetworkPolicyPeer struct {
+	// IPBlock describes the IPAddresses/IPBlocks that is matched in to/from.
+	// IPBlock cannot be set as part of the AppliedTo field
+	// Cannot be set with any other selector.
+	// +optional
+	IPBlock *IPBlock `json:"ipBlock,omitempty"`
+	// Select Pods from AntreaNetworkPolicy's Namespace as workloads in
+	// AppliedTo/To/From fields. If set with NamespaceSelector, Pods are
+	// matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+	// Select all Pods from Namespaces matched by this selector, as
+	// workloads in To/From fields. If set with PodSelector,
+	// Pods are matched from Namespaces matched by the NamespaceSelector.
+	// Cannot be set with any other selector except PodSelector or
+	// ExternalEntitySelector.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// Select ExternalEntities from AntreaNetworkPolicy's Namespace as workloads
+	// in AppliedTo/To/From fields. If set with NamespaceSelector,
+	// ExternalEntities are matched from Namespaces matched by the
+	// NamespaceSelector.
+	// Cannot be set with any other selector except NamespaceSelector.
+	ExternalEntitySelector *metav1.LabelSelector `json:"externalEntitySelector,omitempty"`
```

Comment:
where is ExternalEntitySelector defined?

Reply:
its defined as a label selector in metav1 package.. similar to PodSelector/NamespaceSelector

Reply:
I see. thx

Reply:
Do we want to select ExternalEndpoints later?